x-pack/plugins/maps/public/shared/layers/sources/es_source.js

```diff
@@ -37,6 +37,10 @@ export class AbstractESSource extends AbstractVectorSource {
     return  [this._descriptor.indexPatternId];
   }
 
+  supportsElasticsearchFilters() {
```

Comment:
I'd make this a static function or static property on `Source`, and not a method on the prototype. The return `true/false` is independent of instance-state, but only depends on the type instead.

Reply:
If the method is a static method then it has to be called on the Class instead of the instance. The method is executed from within a layer instance where I have easy access to the source instance, but not the source Class. It would be cumbersome to get the Class of the source instance and then call the static method on the class.

Reply:
`this._source.constructor.supportsElasticsearchFilters`

but yeah, that looks somewhat funny. your call.

Reply:
I will just leave it on prototype since that is consistent with `isFieldAware`, `isRefreshTimerAware`, and others